### PR TITLE
feat: [ISSUE-119] feat: adding a receiver as parameter in the accept swap function

### LINKED ISSUE #119: feat: adding a receiver as parameter in the accept swap function
There is no reason why not allowing to choose the receiver when calling acceptSwap, which is currently setting msg.sender as the receiver of the assets.

Proposal:
- add an address type into the acceptSwap parameter with the name of receiver.
- instead of sending the assets to the msg.sender, send it to the receiver instead.

### PR DESCRIPTION
**Pull request by 0xjoaovpsantos**
This PR close #119 

Proposal in the issue:

- [x] add an address type into the acceptSwap parameter with the name of receiver.
- [x] instead of sending the assets to the msg.sender, send it to the receiver instead.

Plus:

- I created a "approveToken.ts" script to help us to approve tokens to test the flow creating and accepting swaps, it's only a suggestion, I can remove it if you think it's not necessary :)

Tests:

- Contract deployed to test: https://sepolia.etherscan.io/address/0x156A2e49B89127A23a94c3B17C02d13315f06e19

Example transaction: https://sepolia.etherscan.io/tx/0xf9b626c70167d25319884791913857ce748eee1696cb8c35d932ae9ef267f193

- msg.sender send to owner of swap created
- owner of swap created send to receiver

### CODE REVIEW

**Comment by 0xjoaovpsantos** [comment]
I'm fixing the conflicts

**Comment by 0xjoaovpsantos** [comment]
Fixed all conflicts in the PR with main branch and errors in the unit tests  :)

**Comment by 0xjoaovpsantos** [comment]
Okay, now it's ready to review rs

**Comment by 0xjoaovpsantos** [comment]
@0xneves

**Comment by 0xjoaovpsantos** [comment]
Fixed the PR conflicts :)

File: scripts/approveToken.ts
Comment (0xneves) [inline scripts/approveToken.ts:1]: This script is not part of the proposal and should not be committed

File: test/utils/utils.ts
Comment (0xneves) [inline test/utils/utils.ts:1]: This addition is not part of the issue and should be added in this PR

**Review by 0xneves — CHANGES_REQUESTED** [review]
Requesting for changings in the following files:

**Comment by 0xneves** [comment]
@0xjoaovpsantos Happy New Year, how is this task going? We just need minor changes to approve this!

**Comment by 0xjoaovpsantos** [comment]
HEEEEEEEY @0xneves HAPPY NEW YEAR MANNN

Sorry for the delay, I fixed this detail :)

**Review by 0xneves — APPROVED** [review]

**Pull request merged**